registry/remote/credentials/registry.go

```diff
@@ -69,7 +69,7 @@ func Logout(ctx context.Context, store Store, registryName string) error {
 }
 
 // Credential returns a Credential() function that can be used by auth.Client.
-func Credential(store Store) auth.CredentialFunc {
+func Credential(store ReadOnlyStore) auth.CredentialFunc {
```

Comment:
Technically, it's a breaking change since the function signature is changed.

Reply:
The design wasn't good enough. We have to leave this signature change in v3 then.